ISO Unicast Add support for split configuration

Thalley opened the pull request:
Add support for doing a ISO peripheral and a ISO central only configuration. 

fixes https://github.com/zephyrproject-rtos/zephyr/issues/41596

Thalley committed:
Bluetooth: ISO: Add Kconfig for ISO central/peripheral

Add two new Kconfigs: BT_ISO_CENTRAL and BT_ISO_PERIPHERAL
that is used to do central or peripheral only builds,
similar to the BT_CENTRAL or BT_PERIPHERAL Kconfigs.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

Thalley committed:
Bluetooth: ISO: Move functions around to reduce number of guards

Move a few unicast functions around so that we reduce the number
of #if defined guards. No code has changed, but a single
prototype has been added.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

Thalley committed:
Bluetooth: ISO: Change some functions to static

There were a few local functions that were not
defined as static even though they could.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

jhedberg commented on subsys/bluetooth/Kconfig.iso:
```diff
@@ -35,7 +51,6 @@ config BT_ISO_SYNC_RECEIVER
 	select BT_ISO_BROADCAST
 	select BT_OBSERVER
 	select BT_PER_ADV_SYNC
-	select EXPERIMENTAL
```
You shouldn't remove this unless you also remove "[EXPERIMENTAL]" from the option description

cvinayak replied:
Experimental is now added to the parent BT_ISO instead, that should be ok, right?

jhedberg replied:
I thought "[EXPERIMENTAL]" in the option description and `select EXPERIMENTAL` always went hand-in-hand. @carlescufi can you confirm if this is the case or not?

Thalley replied:
I don't mind adding it. 

It's a good question: Should we also have `select EXPERIMENTAL` and `[EXPERIMENTAL]` for hidden Kconfigs, or just for those selectable?

Thalley replied:
Updated the Kconfigs such that we keep a 1:1 between `[EXPERIMENTAL]` and `select EXPERIMENTAL`, effectively removing it from the hidden Kconfigs and adding them to the selectable ones.

Thalley committed:
Bluetooth: ISO: Update ISO Kconfig for EXPERIMENTAL

Ensure that any Kconfig with [EXPERIMENTAL] in the
description, also selects EXPERIMENTAL.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

Thalley committed:
Bluetooth: ISO: Update ISO shell for split config support

The bluetooth ISO shell module now supports the newly
added central and peripheral ISO configs.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

Thalley committed:
Bluetooth: Audio: Remove unicast client iso listen

Remove call to bt_audio_stream_iso_listen which
calls bt_iso_server_register for the unicast client.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

Casper-Bonde-Bose reviewed (dismissed):
LGTM

Thalley commented:
Rebased to solve merge conflict

Thalley commented:
Fixed bad merge causing build error

Thalley commented:
Rebased to fix CI

asbjornsabo reviewed (dismissed):
LGTM

Thalley commented:
Rebased to fix merge conflict

Casper-Bonde-Bose reviewed (dismissed):
Reapprove after rebase

Thalley commented:
Rebased to fix merge conflict

Casper-Bonde-Bose reviewed (approved):
Re-approve after rebase.

mbolivar-nordic commented:
@jhedberg could you please ACK since you are the assignee?
